src/query-builder/aggregate-function-builder.ts

```diff
@@ -159,43 +159,7 @@ export class AggregateFunctionBuilder<DB, TB extends keyof DB, O = unknown>
   }
 
   /**
-   * Adds a `filter` clause with a nested `where exists` clause after the function.
-   *
-   * Similar to {@link WhereInterface}'s `whereExists` method.
-   *
-   * ### Examples
-   *
-   * Count pet owners versus general public:
-   *
-   * ```ts
-   * const result = await db
-   *   .selectFrom('person')
-   *   .select((eb) => [
-   *     eb.fn
-   *       .count<number>('person.id')
-   *       .filterWhereExists((qb) =>
-   *         qb
-   *           .selectFrom('pet')
-   *           .select('pet.id')
-   *           .whereRef('pet.owner_id', '=', 'person.id')
-   *       )
-   *       .as('pet_owner_count'),
-   *     eb.fn.count<number>('person.id').as('total_count'),
-   *   ])
-   *   .executeTakeFirstOrThrow()
-   * ```
-   *
-   * The generated SQL (PostgreSQL):
-   *
-   * ```sql
-   * select count("person"."id") filter(where exists (
-   *   select "pet"."id"
-   *   from "pet"
-   *   where "pet"."owner_id" = "person"."id"
-   * )) as "pet_ower_count",
-   *   count("person"."id") as "total_count"
-   * from "person"
-   * ```
+   * @deprecated Follow [these](https://github.com/koskimas/kysely/releases/tag/0.24.0) instructions to migrate
```

Comment:
My plan is to include a migration guide to the release documentation. From there, we should also link to the upcoming expression recipe.